lib/device_detector/model_extractor.rb

```diff
@@ -0,0 +1,11 @@
+class DeviceDetector
+  class ModelExtractor < VersionExtractor
```

Comment:
This is a bit weird, as it somehow implies that a model is a subtype of version. How about building an abstract `MetadataExtractor`, which both `VersionExtractor` and `ModelExtractor` could inherit from? It so, the method `version_string` would have to be renamed to something more generic.